### ISSUE TITLE
image classification using cnn+nn giving prediction for only single class

### ISSUE THREAD

**Issue by suraj-deshmukh** [fchollet/keras#1869]
I am using two different dataset having 1200 images each. First dataset has 4 classes and second dataset has 6 classes. This is simple image classification problem. But while training, on each epoch I am getting same value for validation accuracy for both the datasets. I have resized all the images of both datasets to 100x100 using imagemagick. 

I don't know where I am making mistake.
Thanks in advance

terminal output : 
Using Theano backend.
Couldn't import dot_parser, loading of dot files will not be possible.
X_train shape: (880, 3, 100, 100)
880 train samples
220 test samples
train:
  0    418
3    179
2    174
1    109
dtype: int64
test:
 0    98
3    55
2    43
1    24
dtype: int64
Train on 880 samples, validate on 220 samples
Epoch 1/5
880/880 [==============================] - 582s - loss: 1.3444 - acc: 0.4500 - val_loss: 1.2752 - val_acc: 0.4455
Epoch 2/5
880/880 [==============================] - 540s - loss: 1.2624 - acc: 0.4750 - val_loss: 1.2802 - val_acc: 0.4455
Epoch 3/5
880/880 [==============================] - 540s - loss: 1.2637 - acc: 0.4750 - val_loss: 1.2712 - val_acc: 0.4455
Epoch 4/5
880/880 [==============================] - 538s - loss: 1.2484 - acc: 0.4750 - val_loss: 1.2623 - val_acc: 0.4455
Epoch 5/5
880/880 [==============================] - 537s - loss: 1.2375 - acc: 0.4750 - val_loss: 1.2486 - val_acc: 0.4455

prediction on test data:
In [26]: model.predict_classes(X_test)
220/220 [==============================] - 37s     
Out[26]: 
array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])

code is attached below:
[ConvNet.txt](https://github.com/fchollet/keras/files/154340/ConvNet.txt)

**Comment by pasky** [comment]
Does it work with a much simpler model?  I'd try just Flatten() + Dense(nb_classes) and it should quickly overfit on the training set.  If not, you likely have problem with data loading; if so, try gradually adding pieces of your original model, tweak the SGD parameters (or try adam) or leave it running longer.

**Comment by suraj-deshmukh** [comment]
@pasky 
I tried with more than 15 epochs but still facing same issue

**Comment by pasky** [comment]
Hope all the other ideas help, then! I didn't expect leaving it to run longer with your current setup would help based on your previous report.
